# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/xul/XULComboboxAccessible.h
# Commit: 1fbca7956efd
# Full Hash: 1fbca7956efd1b73ae3cc50f21ac2780a9e2040c
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2025-04-10 09:29:04
# Regressor Bug: 1549202
# File Overlap Count: 1
# Description:
#   Bug 1549202: Ensure ContentSelectDropdown local accessibles return the focused <select> remote accessible as their parent r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D243029
# ==============================================================================

diff -r 0b9e529a5541 -r 1fbca7956efd accessible/xul/XULComboboxAccessible.h
--- a/accessible/xul/XULComboboxAccessible.h	Wed Apr 09 23:51:01 2025 +0000
+++ b/accessible/xul/XULComboboxAccessible.h	Wed Apr 09 23:51:27 2025 +0000
@@ -37,6 +37,21 @@
   MOZ_CAN_RUN_SCRIPT_BOUNDARY bool AreItemsOperable() const override;
 };
 
+/**
+ * Used for the singular, global instance of a XULCombobox which is rendered
+ * in the parent process and contains the options of the focused and expanded
+ * HTML select in a content document. This combobox should have
+ * id=ContentSelectDropdown
+ */
+class XULContentSelectDropdownAccessible : public XULComboboxAccessible {
+ public:
+  XULContentSelectDropdownAccessible(nsIContent* aContent, DocAccessible* aDoc)
+      : XULComboboxAccessible(aContent, aDoc) {}
+  // Accessible
+
+  virtual Accessible* Parent() const override;
+};
+
 }  // namespace a11y
 }  // namespace mozilla
 
